File: shared/types/evidence.ts
/**
 * Shared types for evidence collection and processing
 */

export interface EvidenceItem {
  id: string
  team_member_id: string
  source: EvidenceSource
  title: string
  description: string
  source_url?: string
  category: EvidenceCategory
  evidence_date: string
  created_at: string
  updated_at?: string
  ai_confidence?: number
  metadata?: Record<string, any>
}

export interface EvidenceItemCreate {
  team_member_id: string
  source: EvidenceSource
  title: string
  description: string
  source_url?: string
  category: EvidenceCategory
  evidence_date: string
  metadata?: Record<string, any>
}

export interface EvidenceFilter {
  team_member_id?: string
  source?: EvidenceSource
  category?: EvidenceCategory
  date_from?: string
  date_to?: string
  search?: string
}

export interface GitLabCommit {
  id: string
  short_id: string
  title: string
  message: string
  author_name: string
  author_email: string
  authored_date: string
  committed_date: string
  web_url: string
  project_id: string
}

export interface GitLabMergeRequest {
  id: string
  title: string
  description: string
  state: string
  author: {
    name: string
    email: string
  }
  created_at: string
  updated_at: string
  merged_at?: string
  web_url: string
  source_branch: string
  target_branch: string
}

export interface JiraTicket {
  id: string
  key: string
  summary: string
  description: string
  status: string
  assignee?: {
    displayName: string
    emailAddress: string
  }
  created: string
  updated: string
  resolved?: string
  project: {
    key: string
    name: string
  }
}

export interface SyncJobStatus {
  id: string
  team_member_id: string
  source: EvidenceSource
  status: 'pending' | 'running' | 'completed' | 'failed'
  progress: number
  started_at?: string
  completed_at?: string
  error_message?: string
  items_processed: number
}

export type EvidenceSource = 'gitlab_commit' | 'gitlab_mr' | 'jira_ticket' | 'document_upload' | 'manual_entry'
export type EvidenceCategory = 'technical' | 'collaboration' | 'delivery' | 'leadership' | 'problem_solving'
export type SyncJobType = 'gitlab' | 'jira' | 'full_sync' 

File: backend/src/models/evidence.py
"""Evidence models for PerformancePulse"""

from typing import Optional, Literal, Dict, Any
from datetime import datetime, date
from pydantic import BaseModel, UUID4, Field, HttpUrl

SourceType = Literal["gitlab_commit", "gitlab_mr", "jira_ticket", "document"]
CategoryType = Literal["technical", "collaboration", "delivery"]

class EvidenceItemBase(BaseModel):
    """Base evidence item model with common fields"""
    title: str = Field(..., min_length=1, max_length=200)
    description: str = Field(..., min_length=1)
    source: SourceType
    category: CategoryType = "technical"
    evidence_date: date
    source_url: Optional[str] = None
    metadata: Dict[str, Any] = Field(default_factory=dict)
    # Author context (optional)
    author_name: Optional[str] = None
    author_email: Optional[str] = None

class EvidenceItemCreate(EvidenceItemBase):
    """Model for creating a new evidence item"""
    team_member_id: UUID4

class EvidenceItemUpdate(BaseModel):
    """Model for updating an existing evidence item"""
    title: Optional[str] = Field(None, min_length=1, max_length=200)
    description: Optional[str] = Field(None, min_length=1)
    category: Optional[CategoryType] = None
    evidence_date: Optional[date] = None
    source_url: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None
    author_name: Optional[str] = None
    author_email: Optional[str] = None

class EvidenceItem(EvidenceItemBase):
    """Complete evidence item model with database fields"""
    id: UUID4
    team_member_id: UUID4
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True 